site/source/tutorials/1010_containers_and_injecting_state/index.md

```diff
@@ -0,0 +1,177 @@
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:1-4 %}
+
+The first import gives the module access to the `Container` factory function which will be used to construct the container. The other imports allow the module to use the `ApplicationContext` to extract state, use the `WorkerContainerProperties` to receive properties from its parent, and the `WorkerContainer` class that will be wrapped by the container.
+
+Next, we need to address the fact that the container has two places to get properties from - its parent widget and the `ApplicationContext`. To tell the container how to manage this, we will create a function called `getProperties`.
+
+{% instruction 'Add the `getProperties` function to the `WorkerContainerContainer` module.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:6-8 %}
+
+{% aside 'Additional injection options' %}
+`Injectors` are capable of injecting more than just properties into a widget. A `getChildren` mapping function can also provided to the `Container`, allowing children to be injected into the widget as well. Just like the `getProperties` function, the `getChildren` function will receive the children from the parent widget and the injector. It then returns the final array of children that will be provided to the contained widget.
+{% endaside %}
+
+The `getProperties` function receives two parameters - an instance of an `Evented` class that holds the state information (`ApplicationContext` in this example) and an object that implements the `WidgetProperties` interface, or one of its descendants (we are using `WorkerContainerProperties`). The first parameter is retrieved from the `Injector` and the second is provided by the widget's parent. The `getProperties` function must then return an object that holds the properties that will be provided to the widget itself. In this example, we are ignoring the properties provided by the parent and returning the `workerData` held by the `ApplicationContext` but more complicated use-cases where both sources are used to generate the properties are also possible.
+
+{% instruction 'Finish the `WorkerContainerContainer` by adding the following code.' %}
+
+{% include_codefile 'demo/finished/biz-e-corp/src/containers/WorkerContainerContainer.ts' lines:10-12 %}
+
+These final lines define the actual `WorkerContainerContainer` class and export it. The `Container` function creates the class by taking in three parameters - the widget's class definition (alternatively, a widget's registry key can be used), the registry key for the `Injector`, and an object literal that provides the mapping functions used to reconcile the two sets of properties and children that the container can receive (one from the `Injector` and one from the parent widget). The returned class is, in fact, a widget (i.e. it descends from `WidgetBase`) and, therefore, can be used just like any other widget.
```

Comment:
"therefore, can be used just like any other widget" -> "therefore may be used just like any other widget"